test/widgets/action_sheet_test.dart

```diff
@@ -1410,6 +1429,196 @@ void main() {
       });
     });
 
+    group('EditButton', () {
+      Future<void> tapEditAndPump(WidgetTester tester, {
+        Duration requestPumpDuration = Duration.zero,
+      }) async {
+        await tester.ensureVisible(find.byIcon(ZulipIcons.edit, skipOffstage: false));
+        await tester.tap(find.byIcon(ZulipIcons.edit));
+        await tester.pump(); // [MenuItemButton.onPressed] called in a post-frame callback: flutter/flutter@e4a39fa2e
+        await tester.pump(requestPumpDuration); // request
+      }
+
+      group('present/absent appropriately', () {
+        /// Test whether the edit-message button is visible, given params.
+        ///
+        /// The message timestamp is 60s before the current time
+        /// ([TestZulipBinding.utcNow]) as of the start of the test run.
+        ///
+        /// The message has streamId: 1 and topic: 'topic'.
+        /// The message list is for that [TopicNarrow] unless [narrow] is passed.
+        void testVisibility(bool expected, {
+          bool self = true,
+          Narrow? narrow,
+          bool allowed = true,
+          int? limit,
+          bool composing = false,
+          bool? errorStatus,
+          bool poll = false,
+        }) {
+          // It's impossible for the compose box to be in editing mode
+          // while, for the same message, the message-edit request is either
+          // in progress or in the error state.
+          assert(!composing || errorStatus == null);
```

Comment:
Oh I see! Yeah this clears up the meaning of `composing`.

I think it is still possible if you initiate the edit in both the channel narrow and the topic narrow containing it, then save the edit in the topic narrow, navigate back and save in the channel narrow, on the same message. If the first request takes long enough to finish. But that's quite an edge case, and is guarded by a containsKey check already.

Reply:
Oh, good find! I can address this edge case in my next revision.